hazelcast/src/test/java/com/hazelcast/cache/CacheTenantUnavailableTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.cache;
+
+import static com.hazelcast.cache.CacheTenantControlTest.setTenantCount;
+import com.hazelcast.cache.impl.HazelcastServerCachingProvider;
+import com.hazelcast.config.CacheConfig;
+import com.hazelcast.config.Config;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import static com.hazelcast.test.HazelcastTestSupport.randomName;
+import com.hazelcast.test.TestHazelcastInstanceFactory;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import com.hazelcast.util.AdditionalServiceClassLoader;
+import com.hazelcast.util.ExceptionUtil;
+import java.io.Serializable;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.HashSet;
+import java.util.Set;
+import javax.cache.Cache;
+import static org.junit.Assert.assertEquals;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+/**
+ * App Server reload / app not loaded tests
+ *
+ * @author lprimak
+ */
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class CacheTenantUnavailableTest extends HazelcastTestSupport {
+    private Config config;
+    private String cacheName;
+    private static final Set<String> disallowClassNames = new HashSet<>();
+
+    @Before
+    public void setup() {
+        cacheName = randomName();
+        config = new Config();
+        ClassLoader configClassLoader = new AdditionalServiceClassLoader(new URL[0],
```

Comment:
Not sure why you need the `AdditionalServiceClassLoader` here. Instead of introducing `SimulateNonExistantClassLoader` I think you can use the `com.hazelcast.internal.util.FilteringClassLoader`. That's the class loader we use in `CacheTypesConfigTest` to test members joining the cluster without KV types on their classpath.

Reply:
```AdditionalServiceClassLoader``` is where tenant control factory resource file is, so I am reusing that.

Reply:
When I am using ```FilteringClassLoader``` I get tenantControl = NoopTenantControl unfortunately, so I think it's easier to use my own very simple class loader